### DISCUSSION TITLE
is there a way to "catch" event when i physically press a note?

### DISCUSSION THREAD

**Discussion by VergilDMC00** [djipco/webmidi#287] [Q&A]
i have a problem with the libraries, i have a physic piano and i would like to know when someone is physically pressing one of the note, is there a way to "catch" that event? I have tried with listening to all channel on the device, but when i phisically press the note it doesnt catch anything while when i "press" the note wit the support of js the event is emitted i am able to catch it

**Comment by djipco** [comment]
What you describe is literally the first example of the [Quick Start guide](https://webmidijs.org/docs/). I suggest you begin with that. 😉

P.S. I moved your post to the "Discussions" section since it does not appera to be a bug with the library.

**Reply by VergilDMC00** [reply]
I did ot like in the quick start, the only difference is that i didnt specify a channel, with this "setting" i catch when i play the device with js, but in the case where i phisicaly press the note the event isnt triggered, is this because of the unspecified channel?

**Comment by djipco** [comment]
You can listen at the device level or at the channel level. Here's how to listen to incoming messages at the device level:

```js
const mySynth = WebMidi.inputs[0]; // First available MIDI input device
mySynth.addListener("noteon", e => console.log(e));
```
Here's how to listen on a specific MIDI channel (10 in this case): 

```js
const mySynth = WebMidi.inputs[0];
mySynth.channels[10].addListener("noteon", e => console.log(e));
```

If you want to view the input devices that are detected on your machine, you can use:

```js
console.log(WebMidi.inputs);
```
